cassandra: fix for missing cassandra tools that are part of distribution

AlexejK opened the pull request:
During work with cassandra I realised that a bundled `cassandra-stress` tool (and a few other tools) were not installed. These tools have slightly different classpath and needed to be handled differently.

This PR should fix the problem.
Also added a couple of extra tests.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

AlexejK commented:
Not sure if `default` build failure is due to my PR or CI setup. It would seem to be related to "failed to read" errors in the test reports?

DomT4 commented:
You can ignore that. It's a Jenkins thing rather than a PR failure per se.

DomT4 commented on Library/Formula/cassandra.rb:
```diff
@@ -86,6 +87,38 @@ def install
+    # Tools use different cassandra.in.sh and should be changed differently
+    move "#{libexec}/tools/bin/cassandra.in.sh", "#{libexec}/tools/bin/cassandra-tools.in.sh"
+    inreplace "#{libexec}/tools/bin/cassandra-tools.in.sh" do |s|
+        # Tools have slightly different path to CASSANDRA_HOME
```
You've indented too hard here. Should be two spaces, this looks like a tab.

DomT4 commented on Library/Formula/cassandra.rb:
```diff
@@ -86,6 +87,38 @@ def install
               "#{share}/cassandra.in.sh"
 
+    # Prepare tools path
+    mkpath "#{libexec}/tools/bin"
```
Use `(libexec/"tools/bin").mkpath`

Same thing but looks nicer.

DomT4 commented on Library/Formula/cassandra.rb:
```diff
@@ -86,6 +87,38 @@ def install
+    mkpath "#{libexec}/tools/bin"
+    rm Dir["#{buildpath}/tools/bin/*.bat"]
+
+    copy Dir["#{buildpath}/tools/bin/*"], "#{libexec}/tools/bin"
```
You should be able to use `(libexec/"tools/bin").install Dir[buildpath/"tools/bin/*"]` I'd imagine here.

DomT4 commented on Library/Formula/cassandra.rb:
```diff
@@ -86,6 +87,38 @@ def install
+    rm Dir["#{buildpath}/tools/bin/*.bat"]
+
+    copy Dir["#{buildpath}/tools/bin/*"], "#{libexec}/tools/bin"
+    copy Dir["#{buildpath}/tools/lib/*.jar"], "#{libexec}/tools"
```
Same notation as above.

DomT4 commented on Library/Formula/cassandra.rb:
```diff
@@ -86,6 +87,38 @@ def install
+    copy Dir["#{buildpath}/tools/lib/*.jar"], "#{libexec}/tools"
+
+    # Tools use different cassandra.in.sh and should be changed differently
+    move "#{libexec}/tools/bin/cassandra.in.sh", "#{libexec}/tools/bin/cassandra-tools.in.sh"
```
Use `mv`.

AlexejK commented:
Thanks @DomT4 , I've updated the PR. For some reason I had 4-space indentation set here, oups!

As for the code, i've updated it with both your changes and also a placement of some files before they are installed. This is mainly because i realised that after adding suggested `.install` commands instead of `copy`/`mv` I would end up with empty useless folders.

Now, most transformations are happening in `buildpath` before things are installed to correct locations. Hope this looks better :)

DomT4 commented:
Not that familiar with Cassandra at all but this is passing CI, so I'll trust this is right :smile:.

Merged in https://github.com/Homebrew/homebrew/commit/c19d7bb5bf909715d2de1ebdc0f3bdb0a282a6bc. Thank you for another contribution to Homebrew @AlexejK! :smiley_cat:
